backup-loop.sh

```diff
@@ -1,25 +1,57 @@
 #!/bin/bash
 
+set -eu
+
 log() {
-  level=$1
+  if [ "$#" -lt 2 ]; then
+    echo "Wrong number of arguments passed to log function" >&2
+    exit 1
+  fi
+  local level=${1}
   shift
   echo "$(date -Iseconds) ${level} $*"
 }
 
-backupSet=${BACKUP_SET}
-excludes="--exclude '*.jar'"
+find_old_backups() {
+  find "${DEST_DIR}" -mtime "+${PRUNE_BACKUPS_DAYS}" "${@}"
+}
+
+backupSet="${BACKUP_SET:-}"
+# shellcheck disable=SC2089
+excludes="${EXCLUDES:-"--exclude '*.jar'"}"
 
-case $TYPE in
+: "${SRC_DIR:=/data}"
+: "${DEST_DIR:=/backups}"
+: "${BACKUP_NAME:=world}"
+: "${INITIAL_DELAY:=120}"
+: "${INTERVAL_SEC:=86400}"
+: "${PRUNE_BACKUPS_DAYS:=7}"
+: "${TYPE:=VANILLA}"
+: "${RCON_PORT:=25575}"
+: "${RCON_PASSWORD:=minecraft}"
+
+case "${TYPE}" in
   FTB|CURSEFORGE)
-    cd ${SRC_DIR}/FeedTheBeast
+    cd "${SRC_DIR}/FeedTheBeast"
+    ;;
+  BUKKIT|SPIGOT|PAPER)
+    cd "${SRC_DIR}"
+    if [ -z "${LEVEL:-}" ]; then
+      LEVEL="world world_nether world_the_end"
+    fi
     ;;
   *)
-    cd ${SRC_DIR}
+    cd "${SRC_DIR}"
     ;;
 esac
 
+: "${LEVEL:=world}"
+
+log INFO "waiting initial delay of ${INITIAL_DELAY} seconds..."
+sleep "$(( INITIAL_DELAY ))"
```

Comment:
...and not sure why the calculated form is needed here

Reply:
It's not needed. I thought that having the ability to specify times in seconds as simple arithmetic expressions is more readable that just some big number, that you have to calculate yourself (if you want to know what given number means and if you want to change the delay).

While more useful for `INTERVAL_SEC`, for consistency reasons I've included it here.

You can now specify in your k8s yaml spec:
```yaml
[...]
    spec:
      containers:
        - name: backup
          image: mc-backup
          env:
            - name: INTERVAL_SEC
              value: "24*60*60" # 24 hours
            - name: INITIAL_DELAY
              value: "3*60" # 3 minutes
[...]
```

Reply:
Only now have I remembered that one can add more human-readable arguments to `sleep`:
```
time sleep 1m 5s

real	1m5.003s
user	0m0.003s
sys	0m0.000s
```
I'll revert the change and make use of implicit splitting, to allow using this :)

Reply:
Good find and thanks, now I see the power in the original change.